### ISSUE TITLE
Upgrade from 2.6.7 to 2.6.9 -> Timestamp errors

### ISSUE THREAD

**Issue by jonnyberthiaume** [chargebee/chargebee-dotnet#18]
Hi,

I've updated from version 2.6.7 -> 2.6.9.

Everything compiled fine and everything seemed fined after a few tests.

Then I received a webhook from ChargeBee and everything went down...

When creating a new webhook event from an IO stream, it works, but when you check the details of the newly created object,  you see an error for the property OccuredAt:

`OccurredAt = 'WebhookEvent.OccurredAt' threw an exception of type 'System.TypeInitializationException'`

This is how I create the webhookEvent object:

```
Public Sub ProcessRequest(context As HttpContext) Implements IHttpHandler.ProcessRequest

	Dim Response As String = Nothing
	Dim ResponseCode As HttpStatusCode = HttpStatusCode.OK

	Try

		' Input
		Using StreamReader As New IO.StreamReader(context.Request.InputStream)

			InputData = StreamReader.ReadToEnd()

		End Using
		If String.IsNullOrWhiteSpace(InputData) Then Throw New ArgumentNullException("The data cannot be null.")

		' Create the webhook event
		WebhookEvent = New ChargeBee.Models.Event(InputData)

		[...]

	Catch ex as Exception
	End Try

	[...]

End Sub
```

I checked and the timestamp is fine in the IO stream (1552915864).

All the modules are up to date, everything else is working fine, I just can't make the webhook object creation works.

If I use 2.6.7, this works flawlessly, in fact it's been working flawlessly all the way back with version 1.x.x.

Please fix for the next release

**Comment by woopata** [comment]
Stumbled upon same issue. Using v2.8.0 and it still fails for any date.

**Comment by cmbankester** [comment]
Still broken in 2.8.5

**Comment by forlayo** [comment]
Still broken in 2.8.7

**Comment by dgvives** [comment]
**Update**

I've developed a small fix for the conversion between Unix TimeStamp and  UTC DateTime. 
Couldn't find how provide a solution to creators, so I pushed the fix to my personal space.  https://github.com/dgvives/chargebee-dotnet/tree/bugfix/patch-timestamp-datetime-conversion 

Hope it helps

**Comment by dgvives** [comment]
**Update**

I've managed to fix this and now on my way to fork and provide the solution I've found to work.

**Comment by dgvives** [comment]
### **[SOLVED]**


 **TL;DR;**

Be sure to call **ChargeBee.Api.ApiConfig.Configure** before accessing DateTime or Timestamp fields to avoid unreasonable Exceptions, even when unit testing or offline.
    
    


**Summary:**
Accessing DateTime and Timestamp fields fail because the timeout for an HttpClientAdapter can't be zero. 

Adding one sentence to @forlayo [comment ](https://github.com/chargebee/chargebee-dotnet/issues/18#issuecomment-737283847) makes it work.

```
        ChargeBee.Api.ApiConfig.Configure("fake-sitename", "fake-apikey");
        var cbEvent = new Event(jsonProvidedAsString);
        var termEnd = cbEvent.Content.Subscription.CurrentTermEnd;
```


ChargeBee Api needs to be initialised before accessing its model. Otherwise it will throw exception when creating **HttpClient** on **ApiUtil** constructor. The same class contains **ConvertFromTimestamp** and **ConverttoTimestamp** methods triggering unexpected errors
https://github.com/chargebee/chargebee-dotnet/blob/1d6934b645c31e9b3e17cc0fbbbf34d582334f0c/ChargeBee/Api/ApiUtil.cs#L21

ApiConfig.ConnectionTimeout is zero until `ChargeBee.Api.ApiConfig(<string>,<string>)`  is run, being first assigned here:
https://github.com/chargebee/chargebee-dotnet/blob/1d6934b645c31e9b3e17cc0fbbbf34d582334f0c/ChargeBee/Api/ApiConfig.cs#L45

Thereby, if **ApiConfig.Configure** is never invoked it would always raise an exception because the timeout for an HttpClientAdapter can't be zero.


Thank you @forlayo on how to reproduce. and  @jonnyberthiaume, @woopata, @cmbankester for also reporting

**Comment by dgvives** [comment]
#27  addresses this.

**Comment by cb-rakesh** [comment]
This has been addressed in v2.10.0
